src/components/molecules/Visualizer/Engine/Cesium/cameraLimiter.ts

```diff
@@ -0,0 +1,237 @@
+      geodsic,
+      length / 2,
+      width / 2,
+    );
+
+    return {
+      cartographicDimensions,
+      cartesianArray,
+    };
+  }, [cesium, property, geodsic]);
+
+  // calculate maximum camera view (outer boundaries)
+  const [cameraViewOuterBoundaries, setCameraViewOuterBoundaries] = useState<
+    Cartesian3[] | undefined
+  >();
+
+  useEffect(() => {
+    const viewer = cesium.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed() || !property?.cameraLimitterTargetArea || !geodsic) return;
+
+    const camera = new CesiumCamera(viewer.scene);
+    camera.setView({
+      destination: Cartesian3.fromDegrees(
+        property.cameraLimitterTargetArea.lng,
+        property.cameraLimitterTargetArea.lat,
+        property.cameraLimitterTargetArea.height,
+      ),
+      orientation: {
+        heading: property?.cameraLimitterTargetArea.heading,
+        pitch: property?.cameraLimitterTargetArea.pitch,
+        roll: property?.cameraLimitterTargetArea.roll,
+        up: camera.up,
+      },
+    });
+
+    const computedViewRectangle = camera.computeViewRectangle();
+    if (!computedViewRectangle) return;
+    const rectangleHalfWidth = Rectangle.computeWidth(computedViewRectangle) * Math.PI * 1000000;
+    const rectangleHalfHeight = Rectangle.computeHeight(computedViewRectangle) * Math.PI * 1000000;
+
+    const {
+      cameraLimitterTargetWidth: width = targetWidth,
+      cameraLimitterTargetLength: length = targetLength,
+    } = property ?? {};
+
+    const { cartesianArray } = calcBoundaryBox(
+      geodsic,
+      length / 2 + rectangleHalfHeight,
+      width / 2 + rectangleHalfWidth,
+    );
+
+    setCameraViewOuterBoundaries(cartesianArray);
+  }, [cesium, property, geodsic]);
+
+  // Manage camera limiter conditions
+  useEffect(() => {
+    const camera = getCamera(cesium?.current?.cesiumElement);
+    const viewer = cesium?.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed() || !property?.cameraLimitterEnabled || !limiterDimensions)
+      return;
+    if (camera) {
+      viewer.camera.setView({
+        destination: getAllowedCameraDestination(camera, limiterDimensions),
+        orientation: {
+          heading: viewer.camera.heading,
+          pitch: viewer.camera.pitch,
+          roll: viewer.camera.roll,
+          up: viewer.camera.up,
+        },
+      });
+    }
+  }, [camera, cesium, property, limiterDimensions]);
+
+  return {
+    limiterDimensions,
+    cameraViewOuterBoundaries,
+    cameraViewBoundariesMaterial,
+  };
+}
+
+export const cameraViewBoundariesMaterial = new PolylineDashMaterialProperty({
+  color: Color.RED,
+});
+
+export type InnerLimiterDimensions = {
+  cartographicDimensions: {
+    rightDemention: Cartographic;
+    leftDemention: Cartographic;
+    topDemention: Cartographic;
+    bottomDemention: Cartographic;
+  };
+  cartesianArray: Cartesian3[];
+};
+
+export const getGeodsic = (
```

Comment:
```suggestion
export const getGeodesic = (
```
!! Don't forget to update wherever this function is being called. !!